[docs] Add demo for Premium (#4750)
mui/mui-x: docs/data/advanced-components/overview.md

The MUI X Pro Plan expands on the limitations of the Community Plan with more advanced features such as multi-filtering, multi-sorting, column resizing and column pinning.

The Pro Plan is available under a commercial license—visit the [Pricing](https://mui.com/pricing/) page for details.
This plan contains the features that are at the limit of what the open-source model can sustain.

> For instance, providing a very comprehensive set of components.

Reviewer: +1 for @joserodolfofreitas's suggestion!

### Premium Plan <span class="plan-premium"></span>

The MUI X Premium Plan contains the most advanced features such as Row grouping, Excel export, Aggregation (🚧), as well as everything that's included in the Pro Plan.

The Premium Plan is available under a commercial license—visit the [Pricing](https://mui.com/pricing/) page for details.
This plan contains highly complex features that can be useful to analyze and group data without the use of an external application.
The price of the plan targets small to medium-size teams.

Premium Plan package:

- [`@mui/x-data-grid-premium`](https://www.npmjs.com/package/@mui/x-data-grid-premium)

The features exclusive to the Premium Plan are marked with the <span class="plan-premium"></span> icon across our documentation.

## Evaluation (trial) licenses

> In accordance with our [End User License Agreement](https://mui.com/store/legal/mui-x-eula/#30-days-evaluation-trial), you can use our Pro and Premium components without a commercial license as long as they are not used for the **development** of a feature intended for production.

Reviewer 2: ```suggestion
In accordance with our [End User License Agreement](https://mui.com/store/legal/mui-x-eula/#30-days-evaluation-trial), you can use the Pro and Premium components without a commercial license for 30 days without restrictions.
```

You do not need to contact us to use these components for evaluation purposes.

> You will need to purchase a commercial license in order to remove the watermarks and console warnings.

Reviewer 2: ```suggestion
You will need to purchase a commercial license in order to remove the watermarks and console warnings, or after the given 30 days period of evaluation.
```

### Premium Plan <span class="plan-premium"></span>

The MUI X Premium Plan contains the most advanced features such as Row grouping, Excel export, Aggregation (🚧), as well as everything that's included in the Pro Plan.

The Premium Plan is available under a commercial license—visit the [Pricing](https://mui.com/pricing/) page for details.
This plan contains highly complex features that can be useful to analyze and group data without the use of an external application.
The price of the plan targets small to medium-size teams.

Premium Plan package:

- [`@mui/x-data-grid-premium`](https://www.npmjs.com/package/@mui/x-data-grid-premium)

The features exclusive to the Premium Plan are marked with the <span class="plan-premium"></span> icon across our documentation.

## Evaluation (trial) licenses

> In accordance with our [End User License Agreement](https://mui.com/store/legal/mui-x-eula/#30-days-evaluation-trial), you can use the Pro and Premium components without a commercial license for 30 days without restrictions.

Reviewer 3: I think that we can keep the same title as before:

```suggestion
In accordance with our [End User License Agreement](https://mui.com/store/legal/mui-x-eula/#evaluation-trial-licenses), you can use the Pro and Premium components without a commercial license for 30 days without restrictions.
```

### Premium Plan <span class="plan-premium"></span>

The MUI X Premium Plan contains the most advanced features such as Row grouping, Excel export, Aggregation (🚧), as well as everything that's included in the Pro Plan.

The Premium Plan is available under a commercial license—visit the [Pricing](https://mui.com/pricing/) page for details.
This plan contains highly complex features that can be useful to analyze and group data without the use of an external application.

> The price of the plan targets small to medium-size teams.

Author: Removed. I copied from the Pro plan.

### Professional support

When purchasing an MUI X Pro or Premium license you get access to professional support for a limited duration.
Support is available on multiple channels, but the recommended channels are:

- GitHub: You can [open a new issue](https://github.com/mui/mui-x/issues/new/choose) and leave your Order ID, so we can prioritize accordingly.

> - Email (only to share private information): You can [open a new issue](https://support.mui.com/hc/en-us/requests/new?tf_360023797420=mui_x) or send an email to x@mui.com.

Reviewer 3: ```suggestion
- Email (**only if your issue requires sharing private information**): You can [open a new issue](https://support.mui.com/hc/en-us/requests/new?tf_360023797420=mui_x) or send an email to x@mui.com.
```

Your OrderId on the issue helps us prioritize the issues based on the following support levels:

1. **MUI X Pro**: MUI's maintainers give these issues more attention than the ones from the Community plan.
2. **MUI X Premium**: Same as MUI X Pro, but with priority over Pro.

> 3. **MUI X Enterprise support Add-on (not available yet)**: A provided SLA with 24h for the first answer.

Reviewer 3: I think that calling it Enterprise will get people confused with our Enterprise plan, not the same. We used this before, would it still work?

```suggestion
3. **MUI X Premium Priority support add-on (not available yet)**: A provided SLA with 24h for the first answer.
```

> 3. **MUI X Enterprise support Add-on (not available yet)**: A provided SLA with 24h for the first answer.

Reviewer 2: Agreed with Matheus, let's leave open until we're clear on this.

```suggestion
3. **MUI X Priority support Add-on (not available yet)**: A provided SLA with 24h for the first answer.
```
